client/src/components/ButtonCreator.tsx

```diff
@@ -0,0 +1,70 @@
+export class ButtonCreator {
+  static createButton(
+    scene,
+    { x, y, width, height, text, onClick, onHover, onOut }
+  ) {
+    // Define button dimensions and position
+    const buttonX = x;
+    const buttonY = y;
+    const buttonWidth = width;
+    const buttonHeight = height;
+
+    // Create a Graphics object for the button
+    let button = scene.add
+      .graphics()
+      .setDepth(1)
+      .fillStyle(0x000000, 0.5) // Default background color
+      .fillRoundedRect(buttonX, buttonY, buttonWidth, buttonHeight, 5)
+      .lineStyle(2, 0xffffff, 1) // Outline
+      .strokeRoundedRect(buttonX, buttonY, buttonWidth, buttonHeight, 5);
+
+    button
+      .setInteractive(
+        new Phaser.Geom.Rectangle(buttonX, buttonY, buttonWidth, buttonHeight),
+        Phaser.Geom.Rectangle.Contains
+      )
+      .setScrollFactor(0);
+
+    let buttonText = scene.add
+      .text(buttonX + buttonWidth / 2, buttonY + buttonHeight / 2, text, {
+        color: "#ffffff",
+        fontSize: "11px",
```

Comment:
If can cutomise the button size lets make font size customisabale also. This is a really useful creation function though.